native/src/oberon/mod.rs

```diff
@@ -152,4 +154,39 @@ impl crate::Oberon {
             token: tkn.to_bytes().to_vec(),
         })
     }
+
+    pub fn verify_token<'a>(request: &VerifyOberonTokenRequest) -> Result<VerifyOberonTokenResponse, Error<'a>> {
+        if request.data.len() == 0 {
+            return Err(Error::InvalidField("must provide data"));
+        }
+
+        let pkbytes: [u8; oberon::PublicKey::BYTES] = match request.pk.as_slice().try_into() {
```

Comment:
There's a `map_err` and `?` syntax that can help safely unwrap and map this, reducing the need to do a match. Cosmetic syntax mostly

```
let pkbytes: [u8; oberon::PublicKey::BYTES] = request
    .pk
    .as_slice()
    .try_into()
    .map_err(|_| Error::InvalidField("invalid public key provided"))?;
```

Reply:
Shall I make that cosmetic change in the other methods inside oberon.rs ?

Reply:
Sure